samples/nrf9160/gnss/src/main.c

```diff
@@ -157,24 +180,32 @@ void lte_disconnect(void)
 }
 #endif /* CONFIG_GNSS_SAMPLE_LTE_ON_DEMAND */
 
-static void agps_data_get(struct k_work *item)
+static void agps_data_get_work_fn(struct k_work *item)
 {
 	ARG_UNUSED(item);
 
 	int err;
 
-	/* With minimal assistance, assistance is injected only when GNSS starts for the
-	 * first time.
+#if defined(CONFIG_GNSS_SAMPLE_ASSISTANCE_SUPL)
+	/* SUPL doesn't usually provide satellite real time integrity information. If GNSS asks
+	 * only for satellite integrity, the request should be ignored.
 	 */
-	if (IS_ENABLED(CONFIG_GNSS_SAMPLE_ASSISTANCE_MINIMAL)) {
-		static bool assistance_done;
-
-		if (assistance_done) {
-			return;
-		}
+	if (last_agps.sv_mask_ephe == 0 &&
+	    last_agps.sv_mask_alm == 0 &&
+	    last_agps.data_flags == NRF_MODEM_GNSS_AGPS_INTEGRITY_REQUEST) {
+		return;
```

Comment:
I did considered that, but because log output somewhat interferes with the default output mode (PVT + NMEA), I decided not to. However, this can happen only once an hour, so maybe that's acceptable. I'll add a message here.